mettle/api/api.go

```diff
@@ -168,6 +169,19 @@ func serve(opts grpcutil.Opts, name string, queueOpts PubSubOpts, apiURL string,
 		numPollers:       queueOpts.NumPollers,
 		deleteJobsTicker: time.NewTicker(10 * time.Minute),
 	}
+
+	// _Technically_ this won't happen more than once in normal running, as we'd only run 1 server, but it does happen in tests.
+	register.Do(func() {
+		prometheus.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+			Namespace: "mettle",
+			Name:      "requests_current",
+		},
+			func() float64 {
+				return float64(len(srv.jobs))
```

Comment:
this could introduce a concurrent map read / modification - if the main logic is updating it at the same time as Prometheus tries to scrape it

Reply:
D'oh. Yeah, brain did not engage on that one.